apps/src/code-studio/assets/DefaultSpritesEditor.jsx

```diff
@@ -38,23 +38,62 @@ export default class DefaultSpritesEditor extends React.Component {
   };
 
   deleteSpriteFromDefaults = spriteName => {
-    let updatedList = {...this.state.defaultList};
-    delete updatedList[spriteName];
+    let updatedList = [...this.state.defaultList];
+    // Find the index at which the sprite should be deleted
+    for (let index = 0; index < updatedList.length; index++) {
+      let sprite = updatedList[index];
+      if (sprite.name === spriteName) {
+        // Item is to be deleted
+        updatedList.splice(index, 1);
+      }
+    }
     this.setState({defaultList: updatedList});
     this.incrementPendingChanges();
   };
 
-  addSpriteToDefaults = (spriteName, spriteCategory) => {
-    let updatedList = {...this.state.defaultList};
-    updatedList[spriteName] = {name: spriteName, key: spriteCategory};
+  addSpriteToDefaults = (addToBeginning, spriteName, spriteCategory) => {
+    let updatedList = [...this.state.defaultList];
+    if (addToBeginning) {
+      updatedList.unshift({name: spriteName, key: spriteCategory});
+    } else {
+      updatedList.push({name: spriteName, key: spriteCategory});
+    }
+    this.setState({defaultList: updatedList});
+    this.incrementPendingChanges();
+  };
+
+  reorderSpriteByOne = (moveForward, spriteName) => {
```

Comment:
Interesting! I went back and for on Forward/Backward and Up/Down. Would Up/Down remove some of the confusion regarding forward/backward?

Reply:
i was wondering the same thing, but i think forward/backward makes more sense... maybe we invert the meaning of forward/backward? (e.g., "forward" actually moves the sprite down the list because it moves the sprite forward in the array)

would that make it clearer, or would it be more confusing for consumers calling this function?

Reply:
I consider moving 'forward' in an array would be moving "up" (towards the beginning) of an array, so I think I would find that more confusing! :laughing: Maybe 'earlier/later'? To imply that you would hit it earlier if you iterated through the array from the beginning?

Reply:
that's fair -- this may be a "me" thing 😄  i like the clarity of `moveForward`, so let's keep as-is